src/Compilers/Core/Portable/Syntax/LineDirectiveMap.cs

```diff
@@ -34,11 +35,11 @@ protected LineDirectiveMap(SyntaxTree syntaxTree)
         {
             // Accumulate all the directives, in source code order
             var syntaxRoot = (SyntaxNodeOrToken)syntaxTree.GetRoot();
-            IEnumerable<TDirective> directives = syntaxRoot.GetDirectives<TDirective>(filter: ShouldAddDirective);
+            IList<TDirective> directives = syntaxRoot.GetDirectives<TDirective>(filter: ShouldAddDirective);
```

Comment:
If we're touching this code, it would be much nicer for these to eb immutable arrays.  changing to this shouldn't be too hard.